File: Proyecto/02. Software/01. Front-End iOS/sell.it.easy_0.0.2/sell.it.easy/sell.it.easy/PropiedadProducto.swift
//
//  PropiedadProducto.swift
//  sell.it.easy
//
//  Created by David Márquez Delgado on 15/8/17.
//  Copyright © 2017 David Márquez Delgado. All rights reserved.
//

import Foundation
import ObjectMapper


class PropiedadProducto : Mappable{

    var id : Int?
    var talla : String?
    var codigoDeBarras : String?
    var codigoDeEmpresa : String?
    var precioCoste : Float?
    var precioVentaPublico : Float?
    var descuento : Float?
    var imagenString : String?
    var IVA : Float?
    
    init(id: Int?, talla: String?, codigoDeBarras: String?, codigoDeEmpresa: String?, precioCoste: Float?, precioVentaPublico: Float?, descuento: Float?, imagenString: String?, IVA: Float?) {
        self.id = id
        self.talla = talla
        self.codigoDeBarras = codigoDeBarras
        self.codigoDeEmpresa = codigoDeEmpresa
        self.precioCoste = precioCoste
        self.precioVentaPublico = precioVentaPublico
        self.descuento = descuento
        self.imagenString = imagenString
        self.IVA = IVA
    }
    
    required init?(map: Map) {}

    func mapping(map: Map) {
        id <- map["id"]
        talla <- map["talla"]
        codigoDeBarras <- map["codigoDeBarras"]
        codigoDeEmpresa <- map["codigoDeEmpresa"]
        precioCoste <- map["precioCoste"]
        precioVentaPublico <- map["precioVentaPublico"]
        descuento <- map["descuento"]
        imagenString <- map["imagenString"]
        IVA <- map["iva"]
    }
}

File: Proyecto/02. Software/02. Back-End Java/workspace/sell.it.easy/src/main/java/com/sell/it/easy/entity/PropiedadProducto.java
package com.sell.it.easy.entity;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Lob;

@Entity
public class PropiedadProducto {

	@Id
	@GeneratedValue(strategy=GenerationType.AUTO)
	private Integer id;
	private String talla;
	private String codigoDeBarras;
	private String codigoDeEmpresa;
	private Float precioCoste;
	private Float precioVentaPublico;
	private Float descuento;
	@Lob
	private String imagenString;
	private Float IVA;
	
	public PropiedadProducto(){}

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getTalla() {
		return talla;
	}

	public void setTalla(String talla) {
		this.talla = talla;
	}

	public String getCodigoDeBarras() {
		return codigoDeBarras;
	}

	public void setCodigoDeBarras(String codigoDeBarras) {
		this.codigoDeBarras = codigoDeBarras;
	}

	public String getCodigoDeEmpresa() {
		return codigoDeEmpresa;
	}

	public void setCodigoDeEmpresa(String codigoDeEmpresa) {
		this.codigoDeEmpresa = codigoDeEmpresa;
	}

	public Float getPrecioCoste() {
		return precioCoste;
	}

	public void setPrecioCoste(Float precioCoste) {
		this.precioCoste = precioCoste;
	}

	public Float getPrecioVentaPublico() {
		return precioVentaPublico;
	}

	public void setPrecioVentaPublico(Float precioVentaPublico) {
		this.precioVentaPublico = precioVentaPublico;
	}

	public Float getDescuento() {
		return descuento;
	}

	public void setDescuento(Float descuento) {
		this.descuento = descuento;
	}

	public String getImagenString() {
		return imagenString;
	}

	public void setImagenString(String imagenString) {
		this.imagenString = imagenString;
	}

	public Float getIVA() {
		return IVA;
	}

	public void setIVA(Float iVA) {
		IVA = iVA;
	}
	
	
}
